spec/services/participants/withdraw/mentor_spec.rb

```diff
@@ -25,7 +25,7 @@
     it "creates a withdrawn state for that user's profile" do
       expect { described_class.call(params: participant_params) }
         .to change { ParticipantProfileState.count }.by(1)
-      expect { User.find(participant_params[:participant_id]).mentor_profile.withdrawn? }
+      expect { ParticipantProfileState.order(created_at: :desc).first.withdrawn? }
```

Comment:
Why? The former mechanism ensure that the user profile has the correct state. That's testing the business logic. The changed one only that the last entry is the expected value, which is testing the implementation.

Reply:
```
it "creates a withdrawn state for that user's profile"
```
Didn't sound quite like what the test was doing.

Fine, I will being back the old logic and rename the test.